File: protenium-iro-hub/app/src/main/java/com/protenium/irohub/ui/start_up/auth/fragment/ForgotPasswordFragment.java
package com.protenium.irohub.ui.start_up.auth.fragment;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProviders;

import com.protenium.irohub.R;
import com.protenium.irohub.base.BaseFragment;
import com.protenium.irohub.ui.start_up.auth.viewmodel.AuthViewModel;

import butterknife.BindView;
import butterknife.ButterKnife;


public class ForgotPasswordFragment extends BaseFragment implements View.OnClickListener {

    private AuthViewModel authViewModel;

    @BindView(R.id.email)
    EditText email;

    @BindView(R.id.send)
    Button send;


    public static ForgotPasswordFragment newInstance() {
        Bundle args = new Bundle();
        ForgotPasswordFragment fragment = new ForgotPasswordFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public View setView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_forgot_password, container, false);
        ButterKnife.bind(this, view);
        authViewModel = ViewModelProviders.of((AppCompatActivity) context).get(AuthViewModel.class);
        authViewModel.getIsLoading().observe(this, isLoading -> {
            if (isLoading) {
                showProgress();
            } else {
                dismissProgress();
            }
        });
        send.setOnClickListener(this);
        return view;
    }

    @SuppressLint("NonConstantResourceId")
    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.send: {
                checkFields();
            }
        }

    }

    private void checkFields() {
        if (TextUtils.isEmpty(send.getText())) {
            email.setError("Email is required");
        }else {
            forgotPassword();
        }
    }

    private void forgotPassword() {
        authViewModel.userForgotPassword(email.getText().toString()).observe(this, response ->{
            if(response.getStatus()){
                Toast.makeText(getContext(), "Forgot Password link sent to mail", Toast.LENGTH_SHORT).show();
            }else {
                Toast.makeText(getContext(), "Something went wrong", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: joyal670-jeetmeet-f4f44707dfe5/app/src/main/java/com/iroid/jeetmeet/start_up/auth/fragment/ForgotPasswordFragment.kt
package com.iroid.jeetmeet.start_up.auth.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.lifecycle.Observer
import com.iroid.jeetmeet.base.BaseFragment
import com.iroid.jeetmeet.databinding.FragmentForgotPasswordBinding
import com.iroid.jeetmeet.start_up.auth.viewmodel.ForgotPasswordViewModel
import com.iroid.jeetmeet.utils.Status
import com.iroid.jeetmeet.utils.Toaster


class ForgotPasswordFragment : BaseFragment()
{
    private lateinit var binding: FragmentForgotPasswordBinding
    private lateinit var forgotPasswordViewModel : ForgotPasswordViewModel

    companion object {

        private const val TAG = "ForgotPasswordFragment"
        private  var user : String? = null

        @JvmStatic
        fun newInstance(type: String) =
            ForgotPasswordFragment().apply {
                arguments = Bundle().apply {
                    putString("user",type)
                }
            }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.let {
            user = it.getString("user")
        }

    }

    override fun setView(
        inflater: LayoutInflater?,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentForgotPasswordBinding.inflate(inflater!!,container,false)
        return binding.root
    }

    override fun initData() {
    }

    override fun setupUI() {

    }

    override fun setupViewModel() {

    }

    override fun setupObserver() {
    }

    override fun onClicks() {

        binding.submitBtn.setOnClickListener {

            val email = binding.etEmail.text.toString()
            if(email.isBlank())
            {
                binding.etEmail.error = "Required"
            }
            else
            {
                setupForgotPasswordObserver(email, user!!)
            }
        }
    }

    private fun setupForgotPasswordObserver(email: String, user: String) {
        forgotPasswordViewModel= ForgotPasswordViewModel()
        forgotPasswordViewModel.forgotPassword(email, user)
        forgotPasswordViewModel.forgotPasswordData.observe(this, Observer {
            when (it.status)
            {
                Status.SUCCESS ->{
                    dismissProgress()

                    Toaster.showToast(requireContext(), it.data!!.message, Toaster.State.SUCCESS, Toast.LENGTH_LONG)
                }
                Status.LOADING ->{
                    showProgress()
                }
                Status.ERROR ->{
                    dismissProgress()
                    Toaster.showToast(
                        requireContext(),
                        it.message!!,
                        Toaster.State.ERROR,
                        Toast.LENGTH_LONG
                    )
                }
                Status.DATA_EMPTY ->{
                    dismissProgress()
                }
                Status.NO_INTERNET ->{
                    dismissProgress()
                    Toaster.showToast(
                        requireContext(),
                        it.message!!,
                        Toaster.State.ERROR,
                        Toast.LENGTH_LONG
                    )
                }
            }
        })
    }

}